### PR TITLE
Address all accessibility errors from WAVE tool

### PR DESCRIPTION
**Pull request by elizhahn**
## What does this PR do (summary of changes)?
 - This addresses all of the accessibility errors after using the lighthouse audit and WAVE tools
 - aria labels were added where necessary
 - headings were provided for all page views
 - icon buttons were provided a title and label
## How should it be tested?
 - fetch the branch down and run a lighthouse audit, accessibility should be at 100%
 - Run the WAVE tool and there should be 0 alerts/errors
## Future Iterations:
 - Add major roles to the page layout
 - add aria properties to describe the state of the lyrics modal

### CODE REVIEW

**Review by m1073496 — APPROVED** [review]
Awesome, looks like we are 100% accessible after these changes. Thanks for your work on this!

**Review by m1073496 — COMMENTED** [review]
Looks like we could use a semantic tag to wrap the whole app. Like a `<main>` or something.

File: public/index.html [hunk 1]
```diff
@@ -2,44 +2,19 @@
...
-    <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
-
-      You can add webfonts, meta tags, or analytics to this file.
-      The build step will place the bundled scripts into the <body> tag.
-
-      To begin the development, run `npm start` or `yarn start`.
-      To create a production bundle, use `npm run build` or `yarn build`.
-    -->
```
Comment (monshan) [inline public/index.html:43]: Thanks for cleaning all this up

File: src/components/App/App.js [hunk 2]
```diff
@@ -63,15 +63,15 @@ class App extends Component {
...
+            <Navigation dynamic="off-home-nav" />
           { !!this.state.mySongs.length &&
           <MySongLibrary
           songs={ this.state.mySongs }
           mySongs={this.state.mySongs }
           handleSong={ this.removeSong }
           buttonIcon={ [<MdRemoveCircle className="handle-song-icon"/>] }
           /> }
-          { this.state.error && <h2>{this.state.error}</h2> }
+          { this.state.error && <p>{this.state.error}</p> }
```
Comment (monshan) [inline src/components/App/App.js:74]: Didn't know that WAVE is so particular with headings, will keep this in mind for the future

File: src/components/SongLibrary/SongLibrary.js [hunk 3]
```diff
@@ -64,10 +64,10 @@ class SongLibrary extends Component {
     return (
       <>
         <SearchBar searchForSongs={this.searchForSongs} />
-        <h2 data-cy="search-msg">{this.state.searchResultsMsg}</h2>
-        <div className="library">
+        <p data-cy="search-msg">{this.state.searchResultsMsg}</p>
+        <main className="library">
           {allSongs}
-        </div>
+        </main>
```
Comment (monshan) [inline src/components/SongLibrary/SongLibrary.js:70]: Glad that we don't get an error from the route-rendering, will implement this in future projects

**Review by monshan — APPROVED** [review]

**Pull request merged**